### ISSUE TITLE
Double best friends cannot be removed

### ISSUE THREAD

**Issue by mark-otaris** [Anaminus/roblox-bug-tracker#542]
In some circumstances that I am not able to reproduce, a user may appear twice in a best friends list, and perhaps also in a normal friends list. If this happens, it is possible to remove the user only once and the other remains there even after clicking on the "Remove Best Friend" button in the dropdown menu.

**Comment by mark-otaris** [comment]
Update: I am now able to reproduce this. If a user appears many times in another's friends list, it is possible to have him appear that many times in the best friends list by adding him as a best friend using the drop down menu for each occurrence of him in the friends list. See [my own best friends list](http://www.roblox.com/Friends.aspx?UserID=28699800#BestFriendsTab) for an example.

**Comment by matthewdean** [comment]
I am saddened that you are trying to delete our friendship @mark-otaris.

**Comment by mark-otaris** [comment]
@matthewdean As a consolation for you, I was also trying to remove everyone else from the list. Only you remained, since you could only be removed once, and I added you back a second time to demonstrate the bug.

**Comment by mark-otaris** [comment]
Best friends no longer exist, so this issue is no longer relevant.
